libauk/src/main/java/com/bitmark/libauk/storage/WalletStorage.kt

```diff
@@ -217,6 +256,19 @@ internal class WalletStorageImpl(private val secureFileStorage: SecureFileStorag
             TransactionEncoder.signMessage(transaction, chainId, credential)
         }
 
+    override fun ethSignTransactionWithIndex(
+        transaction: RawTransaction,
+        chainId: Long,
+        index: Int
+    ): Single<ByteArray> =
+        secureFileStorage.rxSingle { storage ->
+            val json = storage.readOnFilesDir(SEED_FILE_NAME)
```

Comment:
there so many duplicated code here, I think you can create a function named `getETHCredentialWithIndex`